Lib/test/test_tools/test_msgfmt.py

```diff
@@ -240,6 +240,362 @@ def test_strings(self):
                     with self.assertRaises(Exception):
                         msgfmt.make('messages.po', 'messages.mo')
 
+    def test_general_syntax_errors(self):
+        invalid_po_files = (
+            '',
+            '"',
+            '""',
+            '"foo"',
+            # 'msgid',  # invalid but currently accepted
+            'msgstr',
+            'msgid_plural',
+            # 'msgctxt',  # invalid but currently accepted
+            'msgstr',
+            'msgstr[0]',
+            '[0]',
+
+            # unclosed string
+            '''
+msgid "
+msgstr "bar"
+''',
+
+            # unclosed string
+            '''
+msgid "foo
+msgstr "bar"
+''',
+
+            # unclosed string
+            '''
+msgid "foo" "
+msgstr "bar"
+''',
+
+            # unclosed string
+            '''
+msgid "foo"
+"
+msgstr "bar"
+''',
+
+            # illegal backslash
+            '''
+msgid "foo\\"
+"
+msgstr "bar"
+''',
+
+            # msgid with an index
+            '''
+msgid[0] "foo"
+msgstr "bar"
+''',
+
+            # invalid plural index
+            # invalid but currently accepted
+#             '''
+# msgid "foo"
+# msgid_plural "foos"
+# msgstr[foo] "baz"
+# ''',
+
+            # invalid plural index
+            '''
+msgid "foo"
+msgid_plural "foos"
+msgstr[0 "baz"
+''',
+
+            # invalid plural index
+            # invalid but currently accepted
+#             '''
+# msgid "foo"
+# msgid_plural "foos"
+# msgstr[] "baz"
+# ''',
+
+            # invalid plural index
+            '''
+msgid "foo"
+msgid_plural "foos"
+msgstr1] "baz"
+''',
+
+            # invalid plural index
+            '''
+msgid "foo"
+msgid_plural "foos"
+msgstr[[0]] "baz"
+''',
+        )
+        with temp_cwd():
+            for invalid_po in invalid_po_files:
+                with self.subTest(invalid_po=invalid_po):
+                    Path('messages.po').write_text(invalid_po)
+                    # Reset the global MESSAGES dictionary
+                    msgfmt.MESSAGES.clear()
+                    with self.assertRaises((SystemExit, UnboundLocalError,
+                                            IndexError, SyntaxError)):
+                        msgfmt.make('messages.po', 'messages.mo')
```

Comment:
It prints errors on stderr. They should be silenced. Instead, it would be nice to check the output against the expected error message.

Reply:
Updated to capture the output and check error messages :)